gson/src/main/resources/META-INF/proguard/gson.pro

```diff
@@ -59,13 +54,16 @@
   <init>();
 }
 
-# If a class is used in some way by the application and has fields annotated with @SerializedName,
-# keep those fields and the constructors of the class
-# For convenience this also matches classes without no-args constructor, in which case Gson uses JDK Unsafe
-# Based on https://issuetracker.google.com/issues/150189783#comment11
-# See also https://github.com/google/gson/pull/2420#discussion_r1241813541 for a more detailed explanation
+# Keep fields annotated with @SerializedName for classes which are present.
+# If classes with fields annotated with @SerializedName have a no-args
+# constructor keep that as well.
 -if class *
--keepclasseswithmembers,allowobfuscation,allowoptimization class <1> {
```

Comment:
Why did you remove `allowoptimization`?

Reply:
Having `allowoptimization` can allow R8 to propagate field values, so if you have an explicit construction of a serialized class always setting a field to a specific value, then that single value can become the default value for the created instances instead of `0`/`null`.